transaction_decoder/transaction_decoder.py

```diff
@@ -0,0 +1,290 @@
+                tx_metadata = TransactionMetadataTransfer()
+
+                tx_metadata.value = value
+                tx_metadata.properties = TokenTransferProperties()
+                tx_metadata.properties.collection = identifier
+                tx_metadata.properties.identifier = f"{identifier}-{nonce}"
+
+                result.transfers.append(tx_metadata)
+            else:
+                tx_metadata = TransactionMetadataTransfer()
+
+                tx_metadata.value = value
+                tx_metadata.properties = TokenTransferProperties()
+                tx_metadata.properties.collection = identifier
+
+                result.transfers.append(tx_metadata)
+
+        result.sender = metadata.sender
+        result.receiver = receiver
+
+        if len(args) > index:
+            result.function_name = self.hex_to_string(args[index])
+            index += 1
+            result.function_args = args[index:]
+            index += 1
+
+        return result
+
+    def getEsdtTransactionMetadata(
+        self, metadata: TransactionMetadata
+    ) -> Optional[TransactionMetadata]:
+        if metadata.function_name != "ESDTTransfer":
+            return None
+
+        args = metadata.function_args
+        if not args:
+            return None
+
+        if len(args) < 2:
+            return None
+
+        token_identifier = self.hex_to_string(args[0])
+        value = self.hex_to_big_int(args[1])
+
+        result = TransactionMetadata()
+
+        result.sender = metadata.sender
+        result.receiver = metadata.receiver
+        result.transfers = []
+
+        if len(args) > 2:
+            result.function_name = self.hex_to_string(args[2])
+            result.function_args = args[3:]
+
+        tx_metadata = TransactionMetadataTransfer()
+        tx_metadata.value = value
+
+        tx_metadata.properties = TokenTransferProperties()
+        tx_metadata.properties.collection = token_identifier
+        tx_metadata.properties.identifier = token_identifier
+
+        result.transfers.append(tx_metadata)
+        result.value = value
+
+        return result
+
+    def is_address_valid(self, address: str) -> bool:
+        return len(binascii.unhexlify(address)) == 32
+
+    def is_smart_contract_call_argument(self, arg: str) -> bool:
+        if not self.is_hex(arg):
+            return False
+        if len(arg) % 2 != 0:
+            return False
+        return True
+
+    def is_hex(self, value: str) -> bool:
+        try:
+            bytes.fromhex(value)
+            return True
+        except ValueError:
+            return False
+
+    def base64_to_hex(self, str: str) -> str:
+        return binascii.hexlify(base64.b64decode(str)).decode("ascii")
+
+    def hex_to_string(self, hex: str) -> str:
+        return bytes.fromhex(hex).decode("ascii")
+
+    def hex_to_big_int(self, hex: str) -> int:
+        if not hex:
+            return 0
+        return int(hex, 16)
+
+    def base64Encode(self, s: str) -> str:
```

Comment:
Also snake case.

Reply:
It's now deleted since it was not used anywhere.